packages/command-palette/src/use-command-palette.tsx

```diff
@@ -273,6 +283,39 @@ export const useCommandPalette = ( {
 		finalSortedCommands.unshift( viewMySitesCommand );
 	}
 
-	// Return the sorted commands
+	const currentSite = sites.find( ( site ) => site.ID === currentSiteId );
+
+	// If we have a current site and the route includes the site slug, filter and map the commands for single site use.
+	if ( currentSite && currentRoute?.includes( ':site' ) ) {
+		finalSortedCommands = finalSortedCommands.filter( ( command ) =>
+			filterCurrentSiteCommands( currentSite, command )
+		);
+
+		finalSortedCommands = finalSortedCommands.map( ( command: Command ) => {
+			const callback = ( params: CommandCallBackParams ) => {
+				const targetFunction = command?.siteFunctions?.onClick || command.callback;
+
+				return targetFunction( {
+					close: params.close,
+					site: currentSite,
+					setSearch: params.setSearch,
+					setPlaceholderOverride: params.setPlaceholderOverride,
+					command,
+				} );
+			};
+
+			return {
+				name: command.name,
+				label: command.label,
+				...( command.subLabel ? { subLabel: command.subLabel } : {} ),
```

Comment:
This is basically an allow list of which attributes to use. Would it be more future proof (or just more risky) to replace with a denylist of attributes not to extract?

Reply:
I'd say it's pretty much the same. I don't think that the attribute list will grow, and we will eventually throw all this in the garbage (once we implement GB's command palette).